src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -304,6 +314,30 @@ public override string Name
             }
         }
 
+        public override string MetadataName
+        {
+            get
+            {   
+                if (_name.Length != s_defaultNameLength || _name != s_defaultName)
```

Comment:
This seems like a rather indirect way of checking for the `null` name case and causes us to do extra work when the name is `value` (which is pretty common).  Way not add an extra value in packed flags?